src/main/java/software/amazon/smithy/lsp/ext/Document.java

```diff
@@ -81,7 +90,13 @@ public static DocumentPreamble detectPreamble(List<String> lines) {
 
         boolean blankSeparated = lines.get(endOfPreamble).trim().isEmpty();
 
-        return new DocumentPreamble(currentNamespace, namespaceRange, useBlockRange, imports, blankSeparated);
+        return new DocumentPreamble(currentNamespace, namespaceRange, idlVersion, operationInputSuffix,
+                operationOutputSuffix, useBlockRange, imports, blankSeparated);
+    }
+
+    private static Optional<String> getControlStatementValue(String line, String key) {
+        String quotedValue = line.substring(key.length() + 2);
+        return Optional.of(quotedValue.substring(2, quotedValue.length() - 1));
```

Comment:
This needs the second parameter so that the closing quote mark gets removed as well. I did make a small change to combine this into a single substring call though.